.github/workflows/test-build-outputs.yaml

```diff
@@ -0,0 +1,98 @@
+name: test-build-outputs
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+  push:
+    branches:
+      - develop
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  # TODO(devops): extract this into a reusable workflow
+  build-core-osx:
+    name: Build the OSX binaries
+    runs-on: ["self-hosted", "macOS", "X64"]
+    steps:
+      - name: Make checkout speedy
+        run: git config --global fetch.parallel 50
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - run: sudo python3 -m pip install pipenv==2022.6.7
+      - name: Run OSX release script
+        run: ./scripts/osx-release.sh
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: semgrep-osx-${{ github.sha }}
+          path: artifacts.zip
+
+  # TODO(devops): extract this into a reusable workflow
+  build-wheels-osx:
+    runs-on: macos-12
+    needs: [build-core-osx]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: semgrep-osx-${{ github.sha }}
+      - name: Install artifacts
+        run: unzip artifacts.zip
+      - uses: actions/setup-python@v4
+        with:
+          # This is just the Python version to build the wheels
+          python-version: 3.7
+      - name: Build the wheels
+        env:
+          # Relative because build-wheels does a 'cd cli'
+          SEMGREP_CORE_BIN: ../artifacts/semgrep-core
+        run: ./scripts/build-wheels.sh
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: osx-wheel
+          path: cli/dist.zip
+
+  # TODO(devops): extract this into a reusable workflow
+  test-wheels-osx:
+    runs-on: macos-12
+    needs: [build-wheels-osx]
+    steps:
+      - uses: actions/download-artifact@v1
+        with:
+          name: osx-wheel
+      - name: unzip dist
+        run: unzip ./osx-wheel/dist.zip
+      - name: install package
+        run: pip3 install dist/*.whl
+      - name: test version
+        run: semgrep --version
+      - name: e2e semgrep-core test
+        run: echo '1 == 1' | semgrep -l python -e '$X == $X' -
+      - name: e2e spacegrep test
+        run: echo '1 == 1' | semgrep -l generic -e '$X == $X' -
+      - name: test dynamically linked libraries are in /usr/lib/
+        shell: bash {0}
+        run: |
```

Comment:
we should try to minimize the amount of shell code in our workflows because it's tedious to wait for CI to test changes.
Better to add this in a script or in the toplevel Makefile.